src/apps/tezos/layout.py

```diff
@@ -66,6 +70,51 @@ def split_address(address):
     return chunks(address, 18)
 
 
+def split_proposal(proposal):
+    return chunks(proposal, 17)
+
+
 def format_tezos_amount(value):
     formatted_value = format_amount(value, TEZOS_AMOUNT_DIVISIBILITY)
     return formatted_value + " XTZ"
+
+
+async def require_confirm_proposal(ctx, proposals):
+    text = Text("Submit proposal", ui.ICON_SEND, icon_color=ui.PURPLE)
+    text.bold("Proposal:")
+    text.mono(*split_proposal(proposals[0]))
+    await require_confirm(ctx, text, ButtonRequestType.SignTx)
+
+
+async def require_confirm_ballot(ctx, proposal, ballot):
+    text = Text("Submit ballot", ui.ICON_SEND, icon_color=ui.PURPLE)
+    text.bold("Ballot: {}".format(ballot))
+    text.bold("Proposal:")
+    text.mono(*split_proposal(proposal[0]))
+    await require_confirm(ctx, text, ButtonRequestType.SignTx)
+
+
+# use, when there are more then one proposals in one operation
+async def show_proposals(ctx, proposals):
```

Comment:
please merge this with `require_confirm_proposal` (and rename to proposal*s*, probably), instead of making the decision in `sign_tx`

Reply:
also the paginator is smart enough to show only one page when required. you can add a `title` argument to `show_proposal_page` and supply the "proposal/proposals" string this way.

Reply:
Cool, will do that.

Reply:
I looked at the paginator again, and it indeed shows the last page, but it also allows to swipe up. 
So when the user accidentally swipes up a the device raises a FrimwareError, because there is no other page to swipe to. If I recall correctly, this was the reason behind using another screen for a single proposal. 

ISTM the `ui.scroll.change_page` is not handling this edge case, right?

Reply:
that's a bug, but we'll fix it in `change_page`